.travis.yml

```diff
@@ -1,50 +1,46 @@
-language: c
+# Use new container infrastructure to enable caching
 sudo: false
 
+# Choose a lightweight base image; we provide our own build tools.
+language: c
+
+# GHC depends on GMP. You can add other dependencies here as well.
 addons:
   apt:
-    packages: &apt_packages
-      - ghc-ppa-tools
-      - alex-3.1.7
-      - happy-1.19.5
-
-matrix:
-  include:
-    - env: CABALVER=2.0 GHCVER=8.4.1
-      compiler: "ghc-8.4.1"
-      addons: {apt: {packages: [*apt_packages,cabal-install-2.0,ghc-8.4.1], sources: [hvr-ghc]}}
-    - env: CABALVER=2.0 GHCVER=8.2.2
-      compiler: "ghc-8.2.2"
-      addons: {apt: {packages: [*apt_packages,cabal-install-2.0,ghc-8.2.2], sources: [hvr-ghc]}}
-    - env: CABALVER=2.0 GHCVER=8.0.2
-      compiler: "ghc-8.0.2"
-      addons: {apt: {packages: [*apt_packages,cabal-install-2.0,ghc-8.0.2], sources: [hvr-ghc]}}
-    - env: CABALVER=1.24 GHCVER=7.10.3
-      compiler: "ghc-7.10.3"
-      addons: {apt: {packages: [*apt_packages,cabal-install-1.24,ghc-7.10.3], sources: [hvr-ghc]}}
-    - env: CABALVER=1.24 GHCVER=7.8.4
-      compiler: "ghc-7.8.4"
-      addons: {apt: {packages: [*apt_packages,cabal-install-1.24,ghc-7.8.4], sources: [hvr-ghc]}}
-    - env: CABALVER=2.0 GHCVER=head
-      compiler: ": #GHC head"
-      addons: {apt: {packages: [*apt_packages,cabal-install-2.0,ghc-head], sources: [hvr-ghc]}}
-
-  allow_failures:
-   - env: CABALVER=2.0 GHCVER=head
+    packages:
+    - libgmp-dev
+
+# The different configurations we want to test. You could also do things like
+# change flags or use --stack-yaml to point to a different file.
+env:
+- ARGS=""
+#- ARGS="--resolver=nightly"
+- ARGS="--stack-yaml=stack-8.4.1.yaml"
+- ARGS="--stack-yaml=stack-8.2.2.yaml"
+- ARGS="--stack-yaml=stack-8.0.2.yaml"
+- ARGS="--stack-yaml=stack-8.0.1.yaml"
```

Comment:
i usually strive to keep the latest minor version for a given `ghc` major version. i think we can safely ignore `8.0.1` unless there are specific reasons for keeping it